### PR TITLE
improve our `403` error message

### PR DESCRIPTION
**Pull request by adamsachs**
Similar fidesplus PR [here](https://github.com/ethyca/fidesplus/pull/1438)

### Description Of Changes

Small QOL update - improves our `403` error message when an oauth client does not have sufficient permissions for an action to indicate the scopes required for the action, to help troubleshooting and make the API more self-documenting.

Also, while I've been doing some scope troubleshooting, I realized that we don't have a great way of keeping our public facing docs up to date with changes in our app's actual scope registry. For now, I think a reminder on our PR template may be a good way to nudge us to keep those up to date.

### Code Changes

* [x] return the required scopes in the `403` error message sent back to the end user
* [x] small update to our PR template for a reminder that public-facing docs need to be manually updated to reflect new scopes that are added to the app

### Steps to Confirm

* [x] tried an API call for an endpoint that an oauth client i created doesn't have access to - error message looks solid:
```
 curl -i -X 'GET' \
  'http://localhost:8080/api/v1/oauth/client/e511fe11ba99e39b089c65ebc60754e5/scope' \
  -H 'accept: application/json'  -H 'Authorization: Bearer $TOKEN'
HTTP/1.1 403 Forbidden
date: Wed, 22 May 2024 15:25:36 GMT
server: uvicorn
content-length: 77
content-type: application/json
x-ratelimit-limit: 1000
x-ratelimit-remaining: 999
x-ratelimit-reset: 1716391598
retry-after: Wed, 22 May 2024 15:26:38 -0000

{"detail":"Not Authorized for this action. Required scope(s): [client:read]"}
```
In order to do this locally, roughly follow the steps listed [here](https://ethyca.com/docs/dev-docs/api/auth/generating-tokens), i.e.:
- start your local server: `nox -s "build(slim)" "dev(slim)" -- dev`
- go to swagger UI docs at http://localhost:8080/docs, login with root oauth client creds: `fidesadmin`: `fidesadminsecret`
- use the create client `POST /api/v1/oauth/client` endpoint to create a new oauth client, passing in a limited set of scopes in the request body, e.g. 
```
["privacy-request:read"]
```
- copy the `client_id` and `client_secret` of the created client (i.e. in the response body of that API call), and use them as the appropriate form fields in a `POST /api/v1/oauth/token` request, in order to acquire an access token for your new client
- copy the access token, and use it as a bearer auth token in making an API call (this one's probably easier via cURL) that your client does _not_ have the necessary scopes to make, e.g. `GET /api/v1/connection`. Note the error message that you get in your `403` response. it should look like: 
```
 curl -i -X 'GET' \
  'http://localhost:8080/api/v1/connection' \
  -H 'accept: application/json'  -H 'Authorization: Bearer $ACCESS_TOKEN_RETRIEVED_ABOVE'
HTTP/1.1 403 Forbidden
date: Thu, 23 May 2024 18:44:35 GMT
server: uvicorn
content-length: 81
content-type: application/json
x-ratelimit-limit: 1000
x-ratelimit-remaining: 999
x-ratelimit-reset: 1716489936
retry-after: Thu, 23 May 2024 18:45:36 -0000

{"detail":"Not Authorized for this action. Required scope(s): [connection:read]"}
```

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Comment by adamsachs** [comment]
@andres-torres-marroquin - i thought this small tweak in one of our auth overrides may be a nice PR to start diving into our auth layer a bit! let me know if you've got any questions 👍 

there's a corresponding [fidesplus PR](https://github.com/ethyca/fidesplus/pull/1438) too that updates some very rarely used code

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,8 +29,10 @@ The types of changes are:
 - Added developer option to disable auto-initialization on FidesJS bundles. [#4900](https://github.com/ethyca/fides/pull/4900)
 - Adding property ID to served notice history and privacy preference history [#4886](https://github.com/ethyca/fides/pull/4886)
 - Adding privacy_center_config and stylesheet fields to the Property model [#4879](https://github.com/ethyca/fides/pull/4879)
+- Include required scopes in API authorization error (403) messages [#4905](https://github.com/ethyca/fides/pull/4905)
 - Adds generic async callback integration support [#4865](https://github.com/ethyca/fides/pull/4865)
 - Ability to `downgrade` the application DB through the `/admin/db` endpoint [#4893](https://github.com/ethyca/fides/pull/4893)
+- Include the scopes required for a given action in `403` response when client does not have sufficient permissions [#1438](https://github.com/ethyca/fidesplus/pull/1438)
```
Comment (NevilleS) [inline CHANGELOG.md:35]: This looks out of place?

File: src/fides/api/oauth/utils.py [hunk 2]
```diff
@@ -268,7 +268,9 @@ async def verify_oauth_client(
     if not has_permissions(
         token_data=token_data, client=client, endpoint_scopes=security_scopes
     ):
-        raise AuthorizationError(detail="Not Authorized for this action")
+        raise AuthorizationError(
+            detail=f"Not Authorized for this action. Required scope(s): [{', '.join(security_scopes.scopes)}]"
```
Comment (NevilleS) [inline src/fides/api/oauth/utils.py:272]: 👌 thanks, that should help

**Review by NevilleS — APPROVED** [review]
Question in there about the CHANGELOG, but otherwise LGTM thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (adamsachs) [reply CHANGELOG.md:35]: 🤦 trying to move too fast, thanks for catching!

Comment (adamsachs) [reply CHANGELOG.md]: ok fixed in https://github.com/ethyca/fides/pull/4905/commits/0f323ab9f6e36b1eaf46bc15c5758070486f99d4 👍

**Review by andres-torres-marroquin — APPROVED** [review]

**Pull request merged**